### PR TITLE
Add a testcase of SMT and EPT in L1TF

### PR DESCRIPTION
**Pull request by Walterkd**
Since we want to test the smt control in L1TF, we add a testcase of smt in L1TF

- Verification run: http://10.67.132.101/tests/8

### CODE REVIEW

**Comment by Walterkd** [comment]
@wjn740 Please help me review it

File: tests/cpu_bugs/l1tf.pm [hunk 1]
```diff
@@ -51,13 +51,44 @@ my %mitigations_list =
...
+    assert_script_run('echo off>/sys/devices/system/cpu/smt/control');
+    assert_script_run('lscpu | grep "Off-line CPU(s) list"');
+    assert_script_run('cat /sys/devices/system/cpu/smt/active | grep "0"');
+    add_grub_cmdline_settings("l1tf=full,force");
+    grub_mkconfig;
+    Mitigation::reboot_and_wait($self, 150);
+    assert_script_run('lscpu | grep "Off-line CPU(s) list"');
+    assert_script_run('cat /sys/devices/system/cpu/smt/active | grep "0"');
+    assert_script_run('cat /sys/devices/system/cpu/smt/control | grep "forceoff"');
+    my $ret = script_run('echo on>/sys/devices/system/cpu/smt/control');
```
Comment (asmorodskyi) [inline tests/cpu_bugs/l1tf.pm]: not saying that your version is wrong , but mine is smaller :) 
` die  "Control cannot be modified under the mode of forceoff"  unless script_run('echo on>/sys/devices/system/cpu/smt/control'); `

you can still keep yours if you like it more

File: tests/cpu_bugs/l1tf.pm [hunk 2]
```diff
@@ -51,13 +51,44 @@ my %mitigations_list =
...
+    if ($ret eq 0) {
+        record_info('fail', "Control cannot be modified under the mode of forceoff");
+        die;
+    }
+    assert_script_run('cat /sys/devices/system/cpu/smt/control | grep "forceoff"');
+    remove_grub_cmdline_settings("l1tf=full,force");
```
Comment (asmorodskyi) [inline tests/cpu_bugs/l1tf.pm]: this one and at least two lines below look like a candidate to be extracted to a function . You already did grub changes and reboot once before

File: tests/cpu_bugs/l1tf.pm [hunk 3]
```diff
@@ -51,13 +51,44 @@ my %mitigations_list =
...
+    grub_mkconfig;
+    Mitigation::reboot_and_wait($self, 150);
+    assert_script_run('cat /sys/module/kvm_intel/parameters/ept | grep "Y"');
+    my $damn = script_run('modprobe -r kvm_intel | grep "kvm"');
+    if ($damn eq 0) {
+        record_info('fail', "Couldn't find kvm when removed the kvm_intel");
+        die;
+    }
+    assert_script_run('modprobe kvm_intel ept=0;lsmod | grep "kvm"');
+    assert_script_run('cat /sys/module/kvm_intel/parameters/ept | grep "N"');
```
Comment (asmorodskyi) [inline tests/cpu_bugs/l1tf.pm]: again just an advice to increase readability of code : 
```suggestion
    check_param('/sys/module/kvm_intel/parameters/ept',"N"');    
```

where `check_param` : 
```
sub check_param {
   my ($param,$value)= @_;
    assert_script_run("cat $param | grep $value");
}
```

File: tests/cpu_bugs/l1tf.pm (hunk 1 above)
Comment (Walterkd) [reply tests/cpu_bugs/l1tf.pm]: Thanks, already changed

File: tests/cpu_bugs/l1tf.pm (hunk 3 above)
Comment (Walterkd) [reply tests/cpu_bugs/l1tf.pm]: Thanks, already changed

File: tests/cpu_bugs/l1tf.pm (hunk 2 above)
Comment (Walterkd) [reply tests/cpu_bugs/l1tf.pm]: Thanks, already changed

File: tests/cpu_bugs/l1tf.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/cpu_bugs/l1tf.pm]: looks like you forgot to push changes because PR still contains old version

**Pull request merged**